pippy/PipelineStage.py

```diff
@@ -70,6 +70,9 @@ def __init__(
         self.kwargs_chunk_spec = kwargs_chunk_spec
         self.output_chunk_spec = output_chunk_spec
 
+        self.even = torch.cuda.Stream()
+        self.odd = torch.cuda.Stream()
```

Comment:
instead of hard-code even and odd, would it be possible that we let user to set number of streams to use and we have a list of streams?

Reply:
we can make it to have a stream for each chunk. please see the following commit